### PR TITLE
Type acquisition for playground

### PR DESCRIPTION
**Pull request by orta**
wip https://github.com/agentcooper/typescript-play/issues/26 - related https://github.com/microsoft/monaco-typescript/pull/38

### CODE REVIEW

**Comment by orta** [comment]
Works perfect:

- React
- Danger
- Underscore
- All built-in node deps

Works 90%:
- rxjs (could be that transitive relative imports have the wrong path)

Work 10%:
- lodash (could be that the `declare module "../"` isn't hooking up correctly to the root LodashStatic object)
- express (express-serve-static-core routes to the wrong thing)

File: public/main.js [hunk 1]
```diff
@@ -85,6 +76,270 @@ const LibManager = {
...
+      
+      while ((match = es6Pattern.exec(sourceCode)) !== null) {
+        if (match[6]) foundModules.add(match[6])
+      }
+  
+      while ((match = requirePattern.exec(sourceCode)) !== null) {
+        if (match[5]) foundModules.add(match[5])
+      }
+      
+      const moduleJSONURL = (name) => `http://ofcncog2cu-dsn.algolia.net/1/indexes/npm-search/${encodeURIComponent(name)}?x-algolia-agent=Algolia%20for%20vanilla%20JavaScript%20(lite)%203.27.1&x-algolia-application-id=OFCNCOG2CU&x-algolia-api-key=f54e21fa3a2a0160595bb058179bfb1e`
```
Comment (Haroenv) [inline public/main.js]:
```suggestion
      const moduleJSONURL = (name) => `http://ofcncog2cu-dsn.algolia.net/1/indexes/npm-search/${encodeURIComponent(name)}?attributes=types&x-algolia-agent=Algolia%20for%20vanilla%20JavaScript%20(lite)%203.27.1&x-algolia-application-id=OFCNCOG2CU&x-algolia-api-key=f54e21fa3a2a0160595bb058179bfb1e`
```

retrieve only the types this way

Comment (Haroenv) [reply public/main.js]: I created a new API Key for TypeScript's usage: `f8eb0884c44049b48f2e4f6688db299b` (note that for significant usage, we'd want a "powered by Algolia" somewhere on the page :)

**Comment by orta** [comment]
OK, this feels fine on staging 👍

**Pull request merged**